tools/CheckOutputFiles.py

```diff
@@ -0,0 +1,230 @@
+                                    msg="Compare " + str(test_data[index]) +
+                                    " to " + str(expected_data[index]))
+                    else:
+                        self.unit_test.assertEqual(test_data, expected_data)
+                else:
+                    if (test_data.dtype == np.dtype(np.floating) or
+                            test_data.dtype == np.dtype(np.complexfloating)):
+                        for index in np.ndindex(test_data.shape):
+                            with self.unit_test.subTest(index=index):
+                                self.unit_test.assertTrue(
+                                    np.isclose(test_data[index], 0.0, 0.0,
+                                               self.absolute_tolerance))
+                    else:
+                        self.assertTrue(
+                            False,
+                            msg="cannot test non-numeric data without an"
+                            " expected data set to compare against")
+        elif isinstance(h5_file[test_entity], h5py.Group):
+            test_keys = set(h5_file[self.test_entity].keys())
+            checks_passed = True
+            if self.expected_h5_entity is not None:
+                expected_keys = set(h5_file[self.expected_entity].keys())
+                keys_difference = test_keys ^ expected_keys
+                self.unit_test.assertEqual(
+                    keys_difference, {},
+                    "test and expected h5 groups must have identical" +
+                    "subgroups and data members. Found differences: " +
+                    str(keys_difference) + "\nin comparing group " +
+                    test_entity + " to group " + expected_entity)
+                for key in test_keys:
+                    checks_passed = checks_passed and self.check_h5_file(
+                        h5_file, test_entity + "/" + key,
+                        expected_entity + "/" + key)
+            else:
+                for key in test_keys:
+                    checks_passed = checks_passed and self.check_h5_file(
+                        h5_file, test_entity + "/" + key, None)
+            return checks_passed
+
+    def perform_check(self, run_directory):
+        """Apply the h5 check to every h5 file within the `run_directory`
+        that matches the glob `self.h5_glob`.
+
+        If neither the expected nor test objects are present in an h5 file,
+        that file is skipped. However, if none of the matching h5 files have
+        the requested datasets, the test fails, under the assumption that
+        failure to produce any of the anticipated data should be regarded
+        as an error in the executable.
+        """
+        checks_performed = False
+        print(os.path.join(run_directory, self.h5_glob))
+        for filename in glob.glob(os.path.join(run_directory, self.h5_glob)):
+            print(filename)
```

Comment:
Could you replace these `print`s with `logging`, and sprinkle a few more helpful logging message in the control flow to understand what's going on when a test fails? You can look in `tools/CleanOutput.py` for an example how to do the logging and how to set its verbosity.